src/map.cpp

```diff
@@ -5083,15 +5083,11 @@ std::list<item> map::use_charges(const tripoint &origin, const int range,
         // if vehicle has FAUCET can use contents from any of the tanks
         if( v->has_part( "FAUCET" ) ) {
             for( auto &pt : v->parts ) {
-                if( pt.is_tank() ) {
-                    for( const auto &obj : pt.contents() ) {
-                        if( obj.typeId() == type ) {
-                            ret.push_back( pt.drain( quantity ) );
-                            quantity -= ret.back().charges;
-                            if( quantity == 0 ) {
-                                return ret;
-                            }
-                        }
+                if( pt.is_tank() && pt.ammo_current() == type ) {
```

Comment:
That is very much not for me to decide.

Currently, this section allows draining tanks (only) from vehicles that have a faucet, even if not inside the vehicle.

Using vehicle parts that are "pseudo-tools" is covered a little earlier in `map::use_charges(...)`, around line 5057. (It has a nice lambda to avoid unnecessarily verbose checks.) These "pseudo-tools" are available only when inside the vehicle.